astropy/io/fits/column.py

```diff
@@ -1807,7 +1807,8 @@ def _update_column_attribute_changed(self, column, attr, old_value, new_value):
         the numerical index of the column that changed.
         """
         idx = 0
-        for idx, col in enumerate(self.columns):
+        for col in self.columns:
+            idx += 1
```

Comment:
Here the preferred change would be to revert to the original `enumerate` but remove the unnecessary `idx = 0` statement.